### ISSUE TITLE
Conv3d_transpose too slow

### ISSUE THREAD

**Issue by boeseMilch** [tensorflow/tensorflow#19611]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Tested on Arch Linux, OS X and Windows 8
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7 (cpu)
- **Python version**: 2.7.14 and 3.6.5
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: CPU-only issue
- **GPU model and memory**: CPU-only issue
- **Exact command to reproduce**: tf.conv3d_transpose

### Describe the problem
tf.conv3d_transpose seems to be a factor 100 slower than it should be when working with tensorflow on cpu. This was previously reported in Issue #10535 and #7610 , but none have provided a minimal example and both were closed due to inactivity.

### Source code / logs
#Minimal example that shows how much slower conv3d_transposed is compared to einsum
import tensorflow as tf
import time

batch_size = 200
inp = tf.ones((batch_size,28,28,1,1))
filter_conv = tf.ones((28,28,1,1,50))
filter_fc = tf.ones((28,28,1,50))
lat = tf.ones((batch_size,1,1,1,50))

sess = tf.InteractiveSession()
init = tf.global_variables_initializer()
sess.run(init)

t0 = time.time()
sess.run(tf.nn.conv3d_transpose(lat, filter_conv, inp.shape,[1,1,1,1,1],padding='VALID'))
delta_t = time.time() - t0
print delta_t
t0 = time.time()
sess.run(tf.einsum("ijklm,ablm->iablj",lat,filter_fc))
delta_t = time.time() - t0
print delta_t

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ckdotca** [comment]
I'm seeing extreme slowdown for 3D convolutions on CPU as well. I'm going through Keras `Conv3D`, but I suspect it's the same issue. Is there any workaround identified for this for the time being?

**Comment by boeseMilch** [comment]
> I'm seeing extreme slowdown for 3D convolutions on CPU as well. I'm going through Keras Conv3D, but I suspect it's the same issue. Is there any workaround identified for this for the time being?
Yes it might be the same issue. It should only be present in the gradients though, because these use the adjoint of conv3d.
I am unfortunately not aware of any workaround.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ezhulenev** [comment]
@ckdotca @boeseMilch  I've submitted couple of changes that should make Conv3D performance much better on CPU (I'd expecte something like ~5x, maybe ~10x in some cases). One of them is custom kernels for backprop input and filter: https://github.com/tensorflow/tensorflow/commit/e183b8d0328d7398cb6ffc530d1ae8fdbd2111c0

New kernels allocate quite large temporary buffers, so you might see increased peak memory usage. It's possible to fallback on original Eigen kernels using `kernel_label_map` (see https://github.com/tensorflow/tensorflow/blob/8cb0558da924e891aa1bb5d79a6c0c846301e4eb/tensorflow/python/framework/ops.py#L3311), old kernels registered with a "eigen_tensor" label.

I'd be super interested to know how much this helps in your specific case.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by msymp** [comment]
Hi @karmel & @ezhulenev , please advise if we can consider this issue resolved and hence close it. Thanks.

**Comment by msymp** [comment]
Thank you @ezhulenev
